libs/sdk-ui-loaders/src/dashboard/useDashboardLoader.ts

```diff
@@ -0,0 +1,134 @@
+    useCancelablePromise,
+    useWorkspaceStrict,
+} from "@gooddata/sdk-ui";
+import { IDashboardBaseProps } from "@gooddata/sdk-ui-dashboard";
+import isArray from "lodash/isArray";
+import compact from "lodash/compact";
+import { DashboardLoader } from "./dashboardLoader";
+import { DashboardLoadResult, IDashboardLoader, IEmbeddedPlugin } from "./loader";
+
+/**
+ * @alpha
+ */
+export type DashboardLoadStatus = {
+    /**
+     * Status of the dashboard loading.
+     */
+    status: "loading" | "error" | "success";
+
+    /**
+     * Load result - this will be set when status is 'success'. Undefined in all other states.
+     */
+    result?: DashboardLoadResult;
+
+    /**
+     * Load error - this will be set when status is 'error'. Undefined in all other states.
+     */
+    error?: any;
+};
+
+const InitialStatus: DashboardLoadStatus = {
+    result: undefined,
+    error: undefined,
+    status: "loading",
+};
+
+/**
+ * This hook encapsulates load, bootstrap and teardown of a dashboard enhanced by plugins. It is a one-stop
+ * hook to use for React embedding of a Dashboard and when building new dashboard plugins.
+ *
+ * -  The hook can be used in 'dev' mode when developing new dashboard plugins; in this mode, the hook
+ *    expects that it is running inside plugin development toolkit that depends on `@gooddata/sdk-ui-dashboards`
+ *    and includes code for plugin that is under development.
+ *
+ *    In this mode, the hook will combine the dashboard engine with the plugin under development and return
+ *    load result that contains everything needed to mount the dashboard into your application.
+ *
+ * -  The hook can be used in 'prod' mode in order to include a dashboard enhanced with plugins into your
+ *    application.
+ *
+ *    In this mode, the code will dynamically load bundles with the dashboard engine and again
+ *    return load result that contains everything needed to mount the dashboard into your application.
+ *
+ * @param options - load options
+ * @alpha
+ */
+export function useDashboardLoader(options: IDashboardLoadOptions): DashboardLoadStatus {
+    const backend = useBackendStrict(options.backend);
+    const workspace = useWorkspaceStrict(options.workspace);
+    const [loadStatus, setLoadStatus] = useState(InitialStatus);
+    const { dashboard, config, permissions, clientWorkspace, mode, extraPlugins } = options;
+    const baseProps: IDashboardBasePropsForLoader = {
+        backend,
+        workspace,
+        dashboard,
+        config,
+        permissions,
+    };
+
+    const dashboardLoader = useMemo(() => {
+        const extraPluginsArr = isArray(extraPlugins) ? extraPlugins : compact([extraPlugins]);
+        const loader = mode === "dev" ? DashboardLoader.dev() : DashboardLoader.prod();
+
+        initializeLoader(loader, baseProps, extraPluginsArr, clientWorkspace);
+
+        return loader;
+    }, [backend, workspace, dashboard, clientWorkspace, extraPlugins]);
+
+    useCancelablePromise(
```

Comment:
It might be easier to return the stuff returned by the useCancellablePromise, no? You would avoid extra useState and types?